haystack/agents/base.py

```diff
@@ -318,7 +317,7 @@ def on_agent_start(**kwargs: Any) -> None:
             self.callback_manager.on_new_token += lambda token, **kwargs: print_text(token, color=agent_color)
         else:
             self.callback_manager.on_agent_step += lambda agent_step: print_text(
-                agent_step.prompt_node_response, color=agent_color
+                agent_step.prompt_node_response + "\n", color=agent_color
```

Comment:
This change is not related to the streaming but it means that we can slightly change the `print(results)` in the Agent examples, right? For example, here: https://github.com/deepset-ai/haystack/blob/main/examples/agent_multihop_qa.py#L112
Instead of 
```python
result = agent.run(query=question)
print(f"\n{result}")
```
we can do just `print(result)` in the end.

Reply:
@vblagoje If I run the Agent example from this branch, the `\n` is still needed:
```python
result = agent.run(query=question)
print(f"\n{result}")
```
Could you please briefly explain the effect of having the change `end="\n"` in this PR?